### PR TITLE
#130: Optimize Attributes, attempt II

### PR DESCRIPTION
**Pull request by donquixote**
This PR is an improved version of #141.
Not sure why I am creating a new branch for this.

### CODE REVIEW

File: atomium/includes/classes/Attributes.php [hunk 1]
```diff
@@ -51,45 +81,61 @@ public function setAttributes($attributes = array(), $explode = TRUE) {
...
+    if (!isset($this->storage[$name])) {
+      return NULL;
+    }
+    $value = $this->storage[$name];
+    if (\is_bool($value)) {
+      return $value;
+    }
+    unset($value['']);
+    # asort($value);
+    $return = array_values($value);
```
Comment (drupol) [inline atomium/includes/classes/Attributes.php:98]: `return array_values($value);`

Comment (donquixote) [reply atomium/includes/classes/Attributes.php:98]: you are right, probably a leftover.

File: atomium/includes/classes/Attributes.php [hunk 2]
```diff
@@ -306,17 +373,26 @@ public function merge(array $data = array()) {
...
-      });
+    if (\is_bool($actual_value = $this->storage[$key])) {
+      return $actual_value === $value;
+    }
+
+    if (\is_string($value)) {
+      return isset($actual_value[$value]);
+    }
+
+    throw new \InvalidArgumentException("Second parameter is expected to be bool|string.");
```
Comment (drupol) [inline atomium/includes/classes/Attributes.php:395]: I like the idea of this :-)

File: atomium/includes/classes/Attributes.php [hunk 3]
```diff
@@ -136,29 +188,26 @@ public function setAttribute($attribute, $value = FALSE, $explode = TRUE) {
    * @return $this
    */
   public function append($key, $value = FALSE) {
-    $attributes = $this->getStorage();
```
Comment (drupol) [inline atomium/includes/classes/Attributes.php:191]: :face_with_head_bandage:

**Comment by donquixote** [comment]
Obsolete with #183. Closing.

**Pull request closed without merging**